src/Test.Utilities.Sarif/TestUtilitiesExtensions.cs

```diff
@@ -34,5 +36,16 @@ public static TestRuleBehaviors AccessibleWithinContextOnly(this TestRuleBehavio
             // specifies this from the context object that parameterizes the call.
             return behaviors & ~behaviors.AccessibleOutsideOfContextOnly();
         }
+
+        public static IList<T> Shuffle<T>(this IList<T> list, Random random = null)
+        {
+            if (list == null)
+            {
+                return null;
+            }
+
+            random ??= new Random();
```

Comment:
>Random [](http://example.com/codeflow?start=27&length=6)

You can't do this. You must always provide a seed to any Random instantiation. You must always report that seed in any test failure.